packages/server/monitorMemory.ts

```diff
@@ -0,0 +1,32 @@
+import profiler from 'v8-profiler-next'
+import fs from 'fs'
+
+const MB = 2 ** 20
+const monitorMemory = () => {
```

Comment:
A few thoughts on this:

- The production server has `36G` free – we'll have to watch disk space
- It'd be nice to be able to specify a path to where these dumps are written, especially in a future where our blades will have less disk space and we'll probably attach them to a shared, external volume
- It'd be nice if we had two ENV vars:
   - `DEBUG_MEM_MONITOR_ENABLED` which could be true or falsey
   - `DEBUG_MEM_MONITOR_DUMP_PATH`
- It would be nice if the dump file names were prepended with a timestamp, in case we move them to another system for analysis so we would knew what order they were taken in
- It'd be great if we could start and stop the dumps by sending a UNIX signal (see signal events, https://nodejs.org/api/process.html), like `SIGPIPE`

On the later point, then we could start and stop the dumping by doing:

```bash
$ kill -PIPE <pid>
```

Reply:
ohhhh, you're looking to make this a legit part of the codebase? probably a good idea! 

each dump should be < 100MB, so it shouldn't kill us, although true enough, we should watch for it & catch it if it fails & maybe put an upper limit on it.

i'd opt for putting the path in the server code instead of an env variable because anything in the ENV is a PITA. when we need to change it, we can.

yeah, i'll add a timestamp.

i think `DEBUG_MEM_MONITOR_ENABLED` and `SIGPIPE` serve the same purpose, so i'd opt for the latter since it's more powerful.